pkg/tsdb/cloudwatch/services/accounts.go

```diff
@@ -23,20 +24,14 @@ func NewAccountsService(oamClient models.OAMAPIProvider) models.AccountsProvider
 
 func (a *AccountsService) GetAccountsForCurrentUserOrRole(ctx context.Context) ([]resources.ResourceResponse[resources.Account], error) {
 	var nextToken *string
-	sinks := []*oam.ListSinksItem{}
+	sinks := []oam2types.ListSinksItem{}
 	for {
-		response, err := a.ListSinksWithContext(ctx, &oam.ListSinksInput{NextToken: nextToken})
+		response, err := a.ListSinks(ctx, &oam2.ListSinksInput{NextToken: nextToken})
 		if err != nil {
-			var aerr awserr.Error
-			if errors.As(err, &aerr) {
-				switch aerr.Code() {
-				// unlike many other services, OAM doesn't define this error code. however, it's returned in case calling role/user has insufficient permissions
-				case "AccessDeniedException":
-					return nil, fmt.Errorf("%w: %s", ErrAccessDeniedException, aerr.Message())
-				}
+			// TODO: this is a bit hacky, figure out how to do it right in v2
```

Comment:
Do we have a follow up issue to come back to this?

Reply:
We do now! Good call.